auth/useLogin.ts

```diff
@@ -0,0 +1,39 @@
+import { useState, useMemo } from 'react';
+import firebase from 'firebase/app';
+import 'firebase/auth';
+import { AuthHookType } from '../util';
+
+export type loginHook = AuthHookType<
+  firebase.auth.UserCredential,
+  firebase.FirebaseError
+>;
+
+export default (
+  auth: firebase.auth.Auth,
+  email: string,
```

Comment:
I think it would be better to make `email` and `password` variables that are passed into the returned `login` function. This is more in keeping with what some of the other libraries that allow mutation of data, e.g. apollo

Reply:
Actually I was thinking about the exact same thing